src/EventStore.php

```diff
@@ -254,6 +257,39 @@ public function loadEventsByMetadataFrom(StreamName $streamName, array $metadata
     }
 
     /**
+     * @param StreamName[] $streamNames
+     * @param DateTimeInterface|null $since
+     * @param array $metadatas One metadata array per stream name, same index order is required
+     * @return CompositeIterator
+     */
+    public function replay(array $streamNames, DateTimeInterface $since = null, array $metadatas)
+    {
+        if (empty($streamNames)) {
+            throw new \InvalidArgumentException('No stream names given');
+        }
+
+        if (count($streamNames) !== count($metadatas)) {
+            throw new \InvalidArgumentException(sprintf(
+                'One metadata per stream name needed, given %s stream names but %s metadatas',
+                count($streamNames),
+                count($metadatas)
+            ));
+        }
+
+        $iterators = [];
+        foreach ($streamNames as $key => $streamName) {
+            $iterators[] = $this->adapter->replay($streamName, $since, $metadatas[$key]);
+        }
+
+        return new CompositeIterator($iterators, function(Message $message1 = null, Message $message2) {
+            if (null === $message1) {
+                return true;
+            }
+            return (float) $message1->createdAt()->format('U.u') > (float) $message2->createdAt()->format('U.u');
```

Comment:
Damn, or do we need the type cast because PHP does not compare the microseconds correctly?